### PR TITLE
Add required properties to prevAttempts object

### PR DESCRIPTION
**Pull request by sushilmagars**
### Description 
Cypress version `6.2.1` or greater does not forward some of the properties e.g. `duration, title, fullTitle(), context etc` that `mocha-elk-reporter` needs. It throws an error saying - 

`TypeError: test.fullTitle is not a function`

When Cypress version < `6.2.1` - 

```
{
  status: 'passed',
  title: '[uuid:template-test-00001] Passed Test data from reporter',
  fullTitle: 'Unit Test - [uuid:template-test-00001] Passed Test data from reporter',
  duration: 8673,
  err: {},
  platform: 'chromeEmulator',
  browser: 'chrome',
  server: 'jenkins',
  prevAttempts: [
    {
      status: 'Failed',
      title: '[uuid:template-test-00001] Failed Test data from reporter',
      fullTitle: 'Unit Test - [uuid:template-test-00001] Failed Test data from reporter',
      duration: 8673,
      err: {
        stack: 'Sample Failed stack',
        message: 'Sample failed message'
      },
      platform: 'chromeEmulator',
      browser: 'chrome',
      server: 'jenkins'
    },
    {
      status: 'Failed',
      title: '[uuid:template-test-00001] Failed Test data from reporter',
      fullTitle: 'Unit Test - [uuid:template-test-00001] Failed Test data from reporter',
      duration: 8673,
      err: {
        stack: 'Sample Failed stack',
        message: 'Sample failed message'
      },
      platform: 'chromeEmulator',
      browser: 'chrome',
      server: 'jenkins'
    }
  ]
}
```

When Cypress version > `6.2.1` - 

```
{
    status: 'passed',
    title: '[uuid:template-test-00001] Passed Test data from reporter',
    fullTitle: 'Unit Test - [uuid:template-test-00001] Passed Test data from reporter',
    duration: 8673,
    err: {},
    platform: 'chromeEmulator',
    browser: 'chrome',
    server: 'jenkins',
    fullTitle: function() {
      return this.titlePath().join(' ');
    },
    titlePath: function() {
      return this.parent.titlePath().concat([this.title]);
    },
    parent : {
      title: 'Describe block title',
      titlePath: function() {
        var result = [];
        if (this.parent) {
          result = result.concat(this.parent.titlePath());
        }
        if (!this.root) {
          result.push(this.title);
        }
        return result;
      }
    },
    prevAttempts: [
      {
        status: 'Failed',
        err: {
          stack: 'Sample Failed stack',
          message: 'Sample failed message'
        },
        platform: 'chromeEmulator',
        browser: 'chrome',
        server: 'jenkins'
      },
      {
        status: 'Failed',
        err: {
          stack: 'Sample Failed stack',
          message: 'Sample failed message'
        },
        platform: 'chromeEmulator',
        browser: 'chrome',
        server: 'jenkins'
      }
    ]
  }
```

### Solution
With this PR, I'm copying those missing properties from the parent test instance.

### CODE REVIEW

**Comment by mpahuja** [comment]
can we please verify that it does not break non-cypress tests reporting? it should not from what I can see.

I will look into why the CI did not trigger on this PR

**Comment by sushilmagars** [comment]
Okay I'll double check with selenium tests we have. 
Just FYI, I already tested it out with cypress version `6.1.0`, `7.3.0` and `7.5.0`.

**Comment by sushilmagars** [comment]
@mpahuja is it possible to grant me the npm's publish access? I want to create a beta version of these changes, publish them and then use it in my repo's `package.json` that runs the selenium tests against this mocha elk reporter change.

**Comment by mpahuja** [comment]
You not able to npm link it? There is no easy way to add collaborators for
publishing unless we pay for the service as far as I could figure out.

**Comment by sushilmagars** [comment]
Selenium setup is little tricky on my end, don't see a way to test it out locally at least, so was thinking of using beta version. But like you said, I don't think the selenium tests instance have `prevAttempts` property, so this logic -
```
if (failure.prevAttempts && failure.prevAttempts.length) {
```

prevents it from calling the new function I added `addRequiredProps`.

**Comment by sushilmagars** [comment]
@mpahuja, were you able to figure out the issue with CICD. I tried triggering new builds, but its still stuck somewhere it seems. 

image

**Comment by newpablo** [comment]
@mpahuja at this point is cicd the only thing blocking a merge? would be nice to update to latest cypress once this is merged

**Comment by mpahuja** [comment]
@sushilmagars @newpablo apologies for dropping the ball on this. Life happened and I forgot about this PR. For now I have published 2.6.0-beta1 package to unblock you to try out these changes. Will probably wont be able to get to root of the problems and actually verifying if these changes are backward compatible for another 2-3 weeks. But if you can confirm that these changes are backwards compatible then we can just go ahead and merge this PR. Let me know what you think

**Comment by smagar-godaddy** [comment]
@manojnextdoor So was able to use the beta version you posted above and test out both selenium and cypress tests.
And also made sure the its working with Cypress 6.2.1 and 7.7.0 (latest) versions. We should be okay with merge.

-- Sushil

**Review by newpablo — APPROVED** [review]

**Comment by sushilmagars** [comment]
@manojnextdoor i merged the PR, can you please help publish the new version please. Appreciate all the help on this. :)

**Pull request merged**
